libraries/chain/transaction_context.cpp

```diff
@@ -365,6 +368,7 @@ namespace eosio { namespace chain {
       if( ram_delta > 0 ) {
          validate_ram_usage.insert( account );
       }
+      account_ram_delta[account] += ram_delta;
```

Comment:
Instead of adding this line here, you can create a new wrapper function in `apply_context`:

```
void apply_context::add_ram_usage( account_name account, int64_t ram_delta ) {
   trx_context.add_ram_usage( account, ram_delta );
   account_ram_delta[account] += ram_delta;
}
```

Then all calls to `transaction_context::add_ram_usage` in eosio_contract.cpp and apply_context.cpp could be replaced with calls to `apply_context::add_ram_usage`.